src/backend/ticket_service/ticket_service.py

```diff
@@ -376,4 +377,253 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
+            raise HTTPException(status_code=HTTPStatus.NOT_FOUND, detail=f"Comment with comment_id {comment_id} not found for customer {customer_guid}")
+        else:
+            return comment
+    except ValueError as e:
+        logger.error(f"Validation error: {e}")
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail=str(e))
+    except HTTPException as e:
+        raise HTTPException(status_code=HTTPStatus.NOT_FOUND,
+                            detail=f"Comment with comment_id {comment_id} not found for customer {customer_guid}")
+    except Exception as e:
+        if "Database connectivity issue" in str(e):
+            logger.error(f"Database error: {e}")
+            raise HTTPException(
+                status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+                detail="The database is currently unreachable. Please try again later."
+            )
+        logger.error(f"Unexpected error: {e}")
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Internal Server Error")
+
+
+@app.get("/tickets/{ticket_id}/comments", response_model=List[CommentByTicketId], tags=["Comment Management"])
+async def get_comments_by_ticket_id(customer_guid: UUID, ticket_id: str):
+    """Retrieve all comments for a specific ticket_id"""
+    try:
+        comments = db_manager.get_comments_by_ticket_id(str(customer_guid), ticket_id)
+
+        if not comments:
+            logger.info(f"No comments found for ticket_id {ticket_id} and customer {customer_guid}")
+            raise HTTPException(
+                status_code=HTTPStatus.NOT_FOUND,
+                detail=f"No comments found for ticket_id {ticket_id}"
+            )
+        return comments
+    except ValueError as e:
+        logger.error(f"Validation error: {e}")
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail=str(e))
+    except HTTPException as e:
+        logger.info(f"No comments found for ticket_id {ticket_id} and customer {customer_guid}")
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND,
+            detail=f"No comments found for ticket_id {ticket_id}"
+        )
+    except SQLAlchemyError as e:
+        logger.error(f"Database error while retrieving comments: {e}")
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Database error")
+    except Exception as e:
+        if "Database connectivity issue" in str(e):
+            logger.error(f"Database error: {e}")
+            raise HTTPException(
+                status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+                detail="The database is currently unreachable. Please try again later."
+            )
+        logger.error(f"Unexpected error: {e}")
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Internal Server Error")
+
+
+@app.put("/tickets/{ticket_id}/comments/{comment_id}", response_model=CommentUpdateResponse, tags=["Comment Management"])
```

Comment:
This api is not intuitive

Make it /updatecomment

And then add the ticket id and comment_id in the parameters

Reply:
modified sir